### PR TITLE
Gra 987/bug daemon shutdown

### PR DESCRIPTION
**Pull request by abhishek9686**
**This PR should be reviewed after PR #127 is merged**

Testing Steps:- @theguy951357 

1. kill the daemon and check if the daemon shuts down in a reasonable amount of time, it shouldn't take more than a few seconds.
-> you can test by soft killing the daemon `kill -15 <PID>`

### CODE REVIEW

File: nmproxy/nm-proxy.go [hunk 1]
```diff
@@ -13,13 +14,14 @@ import (
 )
 
 // Start - setups the global cfg for proxy and starts the proxy server
-func Start(ctx context.Context, mgmChan chan *nm_models.HostPeerUpdate, stunAddr string, stunPort int) {
+func Start(ctx context.Context, wg *sync.WaitGroup, mgmChan chan *nm_models.HostPeerUpdate, stunAddr string, stunPort int) {
 
 	if config.GetCfg().IsProxyRunning() {
```
Comment (0xdcarns) [inline nmproxy/nm-proxy.go:19]: if you move the wg.Add to below this check, the check for if IsProxyRunning in functions/daemon.go:84 becomes redundant does it not?

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: nmproxy/nm-proxy.go (hunk 1 above)
Comment (abhishek9686) [reply nmproxy/nm-proxy.go:19]: that check is required in functions/daemon.go otherwise context can get overwritten

File: nmproxy/nm-proxy.go [hunk 2]
```diff
@@ -38,3 +40,8 @@ func Start(ctx context.Context, mgmChan chan *nm_models.HostPeerUpdate, stunAddr
 	go manager.Start(ctx, mgmChan)
 	server.NmProxyServer.Listen(ctx)
 }
+
+// IsProxyRunning - fetches the proxy status
```
Comment (mattkasun) [inline nmproxy/nm-proxy.go:44]: why do we need this func; can't one just call config.GetCfg().IsProxyRunning  ?

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: nmproxy/nm-proxy.go (hunk 1 above)
Comment (0xdcarns) [reply nmproxy/nm-proxy.go:19]: okay

File: nmproxy/nm-proxy.go (hunk 2 above)
Comment (0xdcarns) [reply nmproxy/nm-proxy.go:44]: if the func is only called once, I agree, no need

**Review by theguy951357 — APPROVED** [review]
built nodes off of branch
joined network dev
shut down the daemon with kill -15 50481
checked the ps
daemon shut down in a reaasonable time.
pass

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**